tracer/src/Datadog.Trace/Ci/Agent/CIWriterHttpSender.cs

```diff
@@ -0,0 +1,166 @@
+// <copyright file="CIWriterHttpSender.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Datadog.Trace.Agent;
+using Datadog.Trace.Agent.Transports;
+using Datadog.Trace.Ci.Agent.Payloads;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Ci.Agent
+{
+    internal sealed class CIWriterHttpSender : ICIAgentlessWriterSender
```

Comment:
A lot of this logic is essentially the same as the direct log submission (writing to a third party endpoint). I wonder if we could/should combine some of that. For example, I think this could potentially benefit from the circuit breaker work used there (e.g. if the first 3 attempts to write to the endpoint fail, it's probably never going to succeed).

Similarly, we should really look at reducing the default timeouts for requests, they're way too long be default (not specific to this work, applies everywhere, just mentioning it before I forget again!)

Reply:
Let's talk offline about this, I need to understand the direct log submission feature first. And we can handle it in another PR.